services-custom/dynamodb-enhanced/src/test/java/software/amazon/awssdk/enhanced/dynamodb/functionaltests/UpdateBehaviorTest.java

```diff
@@ -145,6 +155,168 @@ public void updateBehaviors_transactWriteItems_secondUpdate() {
+        
+        flattenedMappedTable.putItem(r -> r.item(flattenRecord));
+        
+        NestedRecordWithUpdateBehavior updateNestedRecord = new NestedRecordWithUpdateBehavior();
+        updateNestedRecord.setNestedCounter(100L);
+        
+        CompositeRecord updateCompositeRecord = new CompositeRecord();
+        updateCompositeRecord.setNestedRecord(updateNestedRecord);
+        
+        FlattenRecord updatedFlattenRecord = new FlattenRecord();
+        updatedFlattenRecord.setId("id456");
+        updatedFlattenRecord.setCompositeRecord(updateCompositeRecord);
+        
+        FlattenRecord persistedFlattenedRecord = flattenedMappedTable.updateItem(r -> r.item(updatedFlattenRecord).ignoreNulls(true));
+        
+        assertThat(persistedFlattenedRecord.getCompositeRecord()).isNotNull();
+        assertThat(persistedFlattenedRecord.getCompositeRecord().getNestedRecord().getNestedCounter()).isEqualTo(100L);
+    }
+    
+    @Test
+    public void when_updatingNestedObjectWithMultipleLevelFlattened_existingInformationIsPreserved() {
+
+        NestedRecordWithUpdateBehavior outerNestedRecord = createNestedWithDefaults("id123", 10L);
+        NestedRecordWithUpdateBehavior innerNestedRecord = createNestedWithDefaults("id456", 5L);
+        outerNestedRecord.setNestedRecord(innerNestedRecord);
+
+        CompositeRecord compositeRecord = new CompositeRecord();
+        compositeRecord.setNestedRecord(outerNestedRecord);
+
+        FlattenRecord flattenRecord = new FlattenRecord();
+        flattenRecord.setCompositeRecord(compositeRecord);
+        flattenRecord.setId("id789");
+        
+        flattenedMappedTable.putItem(r -> r.item(flattenRecord));
+        
+        NestedRecordWithUpdateBehavior updateOuterNestedRecord = new NestedRecordWithUpdateBehavior();
+        updateOuterNestedRecord.setNestedCounter(100L);
+        
+        NestedRecordWithUpdateBehavior updateInnerNestedRecord = new NestedRecordWithUpdateBehavior();
+        updateInnerNestedRecord.setNestedCounter(50L);
+        
+        updateOuterNestedRecord.setNestedRecord(updateInnerNestedRecord);
+        
+        CompositeRecord updateCompositeRecord = new CompositeRecord();
+        updateCompositeRecord.setNestedRecord(updateOuterNestedRecord);
+        
+        FlattenRecord updateFlattenRecord = new FlattenRecord();
+        updateFlattenRecord.setCompositeRecord(updateCompositeRecord);
+        updateFlattenRecord.setId("id789");
+        
+        FlattenRecord persistedFlattenedRecord = flattenedMappedTable.updateItem(r -> r.item(updateFlattenRecord).ignoreNulls(true));
+        
+        assertThat(persistedFlattenedRecord.getCompositeRecord()).isNotNull();
+        assertThat(persistedFlattenedRecord.getCompositeRecord().getNestedRecord().getNestedCounter()).isEqualTo(100L);
+        assertThat(persistedFlattenedRecord.getCompositeRecord().getNestedRecord().getNestedRecord().getNestedCounter()).isEqualTo(50L);
+    }
+
```

Comment:
Could you please write a test case to see what happens if we just update a inner nested value

1. Put Item initially
```json
{

  "lastAutoUpdatedOn": "2024-07-26T17:58:24.647273Z",
  "nestedRecord": {
    "nestedTimeAttribute": "2020-05-03T10:00:00Z",
    "nestedRecord": null,
    "id": "id456",
    "nestedUpdateBehaviorAttribute": "testBehaviourAttribute",
    "nestedCounter": 5555,
    "nestedVersionedAttribute": null
  },
  "id": "ignoreNullFalse",
  "version": 1
}
```

Now do a update item to update the inner Nested record like
```java
        NestedRecordWithUpdateBehavior updatedNestedRecord = new NestedRecordWithUpdateBehavior();


        NestedRecordWithUpdateBehavior updatedInnerNestedRecord = new NestedRecordWithUpdateBehavior();
        updatedInnerNestedRecord.setNestedCounter(1L);
        updatedInnerNestedRecord.setId("0001");
        updatedNestedRecord.setNestedRecord(updatedInnerNestedRecord);
        RecordWithUpdateBehaviors update_record = new RecordWithUpdateBehaviors();
        update_record.setId("ignoreNullFalse");
        update_record.setVersion(1L);
        update_record.setNestedRecord(updatedNestedRecord);

        mappedTable.updateItem(r -> r.item(update_record).ignoreNulls(true));

```
Currently I saw that we get 
```
software.amazon.awssdk.services.dynamodb.model.DynamoDbException: The document path provided in the update expression is invalid for update (Service: DynamoDb, Status Code: 400, Request ID: xxx)
```

Reply:
I run 

```
@Test
    public void test() {
        NestedRecordWithUpdateBehavior nestedRecord1 = createNestedWithDefaults("id789", 50L);

        NestedRecordWithUpdateBehavior nestedRecord2 = createNestedWithDefaults("id456", 0L);
        nestedRecord2.setNestedRecord(nestedRecord1);

        RecordWithUpdateBehaviors record = new RecordWithUpdateBehaviors();
        record.setId("id123");
        record.setNestedRecord(nestedRecord2);

        mappedTable.putItem(record);

        NestedRecordWithUpdateBehavior updatedNestedRecord2 = new NestedRecordWithUpdateBehavior();
        long innerNestedCounter = 100L;
        updatedNestedRecord2.setNestedCounter(innerNestedCounter);

        // A Null Attribute set as part of update
        updatedNestedRecord2.setAttribute(TEST_ATTRIBUTE);

        NestedRecordWithUpdateBehavior updatedNestedRecord1 = new NestedRecordWithUpdateBehavior();
        updatedNestedRecord1.setNestedRecord(updatedNestedRecord2);
        long outerNestedCounter = 200L;
        updatedNestedRecord1.setNestedCounter(outerNestedCounter);

        RecordWithUpdateBehaviors update_record = new RecordWithUpdateBehaviors();
        update_record.setId("id123");
        update_record.setVersion(1L);
        update_record.setNestedRecord(updatedNestedRecord1);

        mappedTable.updateItem(r -> r.item(update_record).ignoreNulls(true));

        RecordWithUpdateBehaviors persistedRecord = mappedTable.getItem(r -> r.key(k -> k.partitionValue("id123")));

        verifyMultipleLevelNestingTargetedUpdateBehavior(persistedRecord, outerNestedCounter, innerNestedCounter);

        assertThat(persistedRecord.getNestedRecord().getNestedRecord().getAttribute()).isEqualTo(
            TEST_ATTRIBUTE);

    }
```

and I get a successful. In the test case specified, a new level of nesting is being created with an update. Our existing behaviour does not support it.

Reply:
I will add this added check as part of the existing test though. 

```
updatedNestedRecord2.setAttribute(TEST_ATTRIBUTE);
assertThat(persistedRecord.getNestedRecord().getNestedRecord().getAttribute()).isEqualTo(
            TEST_ATTRIBUTE);
```

Reply:
Added test for null and empty string attribute setting in the next revision

Reply:
```java
    DynamoDbClient myDynamoDbClient = DynamoDbClient.create();
    String MY_TABLE = "joviegas-test-update_07cfdb74-8757-4918-9a05-baeb6eae1682";


    private DynamoDbTable<RecordWithUpdateBehaviors> getTheTable() {


        DynamoDbEnhancedClient enhancedClient = DynamoDbEnhancedClient.builder()
                                                                      .dynamoDbClient(myDynamoDbClient).extensions(
                Stream.concat(ExtensionResolver.defaultExtensions().stream(),
                              Stream.of(AutoGeneratedTimestampRecordExtension.create())).collect(Collectors.toList())).build();


        DynamoDbTable<RecordWithUpdateBehaviors> mappedTable =
            enhancedClient.table(MY_TABLE, TABLE_SCHEMA);
        return mappedTable;
    }



   @Test
    public void when_updatingNestedObjectWithSingleLevel_() {


        DynamoDbTable<RecordWithUpdateBehaviors> mappedTable = getTheTable();

        String uuid = UUID.randomUUID().toString();


        NestedRecordWithUpdateBehavior nestedRecord = createNestedWithDefaults("id456", 5555L);

        RecordWithUpdateBehaviors insertRecord = new RecordWithUpdateBehaviors();
        insertRecord.setId(uuid);
        insertRecord.setNestedRecord(nestedRecord);

        mappedTable.putItem(insertRecord);

        NestedRecordWithUpdateBehavior updatedNestedRecord = new NestedRecordWithUpdateBehavior();


        NestedRecordWithUpdateBehavior updatedInnerNestedRecord = new NestedRecordWithUpdateBehavior();
        updatedInnerNestedRecord.setNestedCounter(0L);
        updatedInnerNestedRecord.setId("0000");

        updatedNestedRecord.setNestedUpdateBehaviorAttribute(null);
        updatedNestedRecord.setNestedRecord(updatedInnerNestedRecord);
        // updatedNestedRecord.setNestedCounter(9999L);

        RecordWithUpdateBehaviors update_record = new RecordWithUpdateBehaviors();
        update_record.setId(uuid);
        update_record.setVersion(1L);
        update_record.setNestedRecord(updatedNestedRecord);

        mappedTable.updateItem(r -> r.item(update_record).ignoreNulls(true));


        GetItemResponse item = myDynamoDbClient.getItem(f -> f.tableName(MY_TABLE).key(Collections.singletonMap("id",
                                                                                                                AttributeValue.builder().s(uuid).build())));

        System.out.println("item is " +item.item());

        // verifySingleLevelNestingTargetedUpdateBehavior(persistedRecord, updatedNestedCounter);

    }
```

Could you please try this on real Dynamo table the test case in this class used LocalDDB which is a mock DDB.
Might be service behaves differently for this .

If we confirm its not a SDK issue then we can raise issue on DDB team internally